src/screens/Departures/state/stop-place-details-state.ts

```diff
@@ -0,0 +1,135 @@
+import {ErrorType, getAxiosErrorType} from '@atb/api/utils';
+import {getStopsDetails} from '@atb/api/departures/stops-nearest';
+import {StopsDetailsQuery} from '@atb/api/types/generated/StopsDetailsQuery';
+
+export type StopsDetailsDataState = {
+  data: StopsDetailsQuery | null;
+  error?: {type: ErrorType};
+  ids?: Array<string>;
+  isLoading: boolean;
+};
+
+const initialState: StopsDetailsDataState = {
+  data: null,
+  error: undefined,
+  ids: undefined,
+  isLoading: false,
+};
+
+type StopsDetailsDataActions =
+  | {
+      type: 'LOAD_DETAILS';
+      locations?: Array<string>;
+    }
+  | {
+      type: 'STOP_LOADER';
+    }
+  | {
+      type: 'UPDATE_STOP_DETAILS';
+      ids?: Array<string>;
+      result: StopsDetailsQuery;
+    }
+  | {
+      type: 'SET_ERROR';
+      error: ErrorType;
+    };
+
+const reducer: ReducerWithSideEffects<
+  StopsDetailsDataState,
+  StopsDetailsDataActions
+> = (state, action) => {
+  switch (action.type) {
+    case 'LOAD_DETAILS': {
+      if (!action.locations) return NoUpdate();
+      return UpdateWithSideEffect<
+        StopsDetailsDataState,
+        StopsDetailsDataActions
+      >(
+        {
+          ...state,
+          isLoading: true,
+          error: undefined,
+        },
+        async (state, dispatch) => {
+          if (!action.locations) return;
+          try {
+            const result = await getStopsDetails({ids: action.locations});
+            dispatch({
+              type: 'UPDATE_STOP_DETAILS',
+              ids: action.locations,
+              result,
+            });
+          } catch (e) {
+            dispatch({
+              type: 'SET_ERROR',
+              error: getAxiosErrorType(e),
+            });
+          } finally {
+            dispatch({type: 'STOP_LOADER'});
+          }
+        },
+      );
+    }
+
+    case 'STOP_LOADER': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+      });
+    }
+
+    case 'UPDATE_STOP_DETAILS': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+        ids: action.ids,
+        data: action.result,
+      });
+    }
+
+    case 'SET_ERROR': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        error: {
+          type: action.error,
+        },
+        data: state.data,
+      });
+    }
+
+    default:
+      return NoUpdate();
+  }
+};
+
+/***
+ * Use state for fetching details for a list of stopPlaces
+ *
+ * @param {Array<string>} [locationIds] - NSR ids to fetch details for
+ */
+export function useStopsDetailsData(locationIds?: Array<string>) {
+  const [state, dispatch] = useReducerWithSideEffects(reducer, initialState);
+
+  const refresh = useCallback(
+    () =>
+      dispatch({
+        type: 'LOAD_DETAILS',
+        locations: locationIds,
+      }),
+    [locationIds],
+  );
+
+  useEffect(refresh, [JSON.stringify(locationIds)]);
+
+  return {
+    state,
+    refresh,
+  };
```

Comment:
Her returnerer man `refresh`-funksjon, som jeg ikke ser blir brukt noe sted. Samt at i `state` ligger også felter som `error`, `ids` og `isLoading` som per nå ikke blir brukt. Er det noen planer for å bruke denne reduceren andre steder senere hvor dette vil bli tatt i bruk, ellers tenker jeg det er greit å ta bort.